core/modules/replica/DatabaseMySQLRow.cc

```diff
@@ -0,0 +1,174 @@
+
+bool Row::get(size_t             columnIdx,  std::string& value) const { return ::getAsString(*this, columnIdx,  value); }
+bool Row::get(std::string const& columnName, std::string& value) const { return ::getAsString(*this, columnName, value); }
+
+bool Row::get(size_t columnIdx, uint64_t& value) const { return ::getAsNumber(*this, columnIdx, value); }
+bool Row::get(size_t columnIdx, uint32_t& value) const { return ::getAsNumber(*this, columnIdx, value); }
+bool Row::get(size_t columnIdx, uint16_t& value) const { return ::getAsNumber(*this, columnIdx, value); }
+bool Row::get(size_t columnIdx, uint8_t&  value) const { return ::getAsNumber(*this, columnIdx, value); }
+
+bool Row::get(std::string const& columnName, uint64_t& value) const { return ::getAsNumber(*this, columnName, value); }
+bool Row::get(std::string const& columnName, uint32_t& value) const { return ::getAsNumber(*this, columnName, value); }
+bool Row::get(std::string const& columnName, uint16_t& value) const { return ::getAsNumber(*this, columnName, value); }
+bool Row::get(std::string const& columnName, uint8_t&  value) const { return ::getAsNumber(*this, columnName, value); }
+
+bool Row::get(size_t columnIdx, int64_t& value) const { return ::getAsNumber(*this, columnIdx, value); }
+bool Row::get(size_t columnIdx, int32_t& value) const { return ::getAsNumber(*this, columnIdx, value); }
+bool Row::get(size_t columnIdx, int16_t& value) const { return ::getAsNumber(*this, columnIdx, value); }
+bool Row::get(size_t columnIdx, int8_t&  value) const { return ::getAsNumber(*this, columnIdx, value); }
+
+bool Row::get(std::string const& columnName, int64_t& value) const { return ::getAsNumber(*this, columnName, value); }
+bool Row::get(std::string const& columnName, int32_t& value) const { return ::getAsNumber(*this, columnName, value); }
+bool Row::get(std::string const& columnName, int16_t& value) const { return ::getAsNumber(*this, columnName, value); }
+bool Row::get(std::string const& columnName, int8_t&  value) const { return ::getAsNumber(*this, columnName, value); }
+
+bool Row::get(size_t columnIdx, float&  value) const { return ::getAsNumber(*this, columnIdx, value); }
+bool Row::get(size_t columnIdx, double& value) const { return ::getAsNumber(*this, columnIdx, value); }
+
+bool Row::get(std::string const& columnName, float&  value) const { return ::getAsNumber(*this, columnName, value); }
+bool Row::get(std::string const& columnName, double& value) const { return ::getAsNumber(*this, columnName, value); }
+
+bool Row::get(size_t columnIdx, bool& value) const {
+    uint8_t number;
+    if (::getAsNumber(*this, columnIdx, number)) {
+        value = number != '0';
+        return true;
+    }
+    return false;
+}
+
+bool Row::get(std::string const& columnName, bool&  value) const {
+    uint8_t number;
+    if (::getAsNumber(*this, columnName, number)) {
+        value = number != '0';
```

Comment:
SAme here

Reply:
This is for database type 'BOOLEAN' which in MySQL is returned by the C language library as an 8-bit integer/character set to '0' as a representation of the false values. Hence this comparison. The 'BIT(1)' type is to covered by this implementation.